### ISSUE TITLE
Support other schematic file types

### ISSUE THREAD

**Issue by plasmatech8** [The-Redstone-Index/redstone-index#9]
Support converting to/from NBT and other file formats to allow users to upload/download Redstone schematics in their prefered format.

File types:
* World folders
* Litematica
* Schem

NBT would likely need to be the primary storage format since that is what the deepslate library uses to render structures.

**Comment by plasmatech8** [comment]
Relevant: 
* https://github.com/misode/vscode-nbt/blob/master/src/editor/Schematics.ts
* https://github.com/misode/vscode-nbt/blob/ac96c324b3c96e5daadca03b41a24566d72dd967/src/NbtDocument.ts#L205

Apparently the [vscode-nbt](https://github.com/misode/vscode-nbt) extension can read and render litematic files. Don't know exactly if it supports file conversions and how to fit it into the application.

**Comment by EndingCredits** [comment]
This might be useful for litematic support:
https://endingcredits.github.io/litematic-viewer/

**Comment by plasmatech8** [comment]
> This might be useful for litematic support: https://endingcredits.github.io/litematic-viewer/

Thanks @EndingCredits. Some of the code could be useful for file type converters. :)

I have basically put this project on hold... -

But do you think it would be better if I used litematic files as the base storage format (stored in the object bucket) or keep using NBT files?

**Comment by EndingCredits** [comment]
I think more people use litematic, but then it depends what the target audience is.

**Comment by EndingCredits** [comment]
Is there any plans to pick this up again? I think it's a cool project and already 90% of the way there, just needs some final things.

I think there are a few people in the TMC community who would be interested in helping out if you are interested.

**Comment by plasmatech8** [comment]
@EndingCredits Yeah sorry. I don't have specific plans continue. I don't know how or have time to implement the final two features, and I am uncertain of how much interest there will be in using the project. If someone were to build a Minecraft server + plugin to upload/download schematics with the platform (via the REST API), that might make things more interesting.

If people in TMC community express interest and are willing to chip in, I would be happy to continue.

I guess I could also deploy a production version of the application as it stands currently if we wanted. It would just be missing the two features which would make it less usable until they are implemented.
